### PR TITLE
LLM: check user env

### PR DESCRIPTION
**Pull request by WeiguangHan**
@hkvision Check user env.

### CODE REVIEW

**Comment by WeiguangHan** [comment]
A sample output are as follows.
```
-----------------------------------------------------------------
PYTHON_VERSION=3.9.18
-----------------------------------------------------------------
transformers=4.34.0
-----------------------------------------------------------------
torch=2.1.0a0+cxx11.abi
-----------------------------------------------------------------
BigDL Version: 2.5.0b20240317
-----------------------------------------------------------------
/home/arda/anaconda3/envs/weiguang-llm-test/lib/python3.9/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: ''If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
  warn(
ipex=2.1.10+xpu
-----------------------------------------------------------------
CPU Information: 
Architecture:                    x86_64
CPU(s):                          32
Model name:                      13th Gen Intel(R) Core(TM) i9-13900K
CPU max MHz:                     5800.0000
CPU min MHz:                     800.0000
-----------------------------------------------------------------
Total Memory: 62.5479 GB
-----------------------------------------------------------------
Operating System: 
Ubuntu 22.04.3 LTS \n \l

-----------------------------------------------------------------
03:00.0 VGA compatible controller: Intel Corporation Device 56a0 (rev 08) (prog-if 00 [VGA controller])
        Subsystem: Device 1ef7:1307
        Flags: bus master, fast devsel, latency 0, IRQ 201
        Memory at 84000000 (64-bit, non-prefetchable) [size=16M]
        Memory at 4000000000 (64-bit, prefetchable) [size=16G]
        Expansion ROM at 85000000 [disabled] [size=2M]
        Capabilities: <access denied>
        Kernel driver in use: i915
        Kernel modules: i915
-----------------------------------------------------------------
Linux arda-arc11 5.19.0-41-generic #42~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Apr 18 17:40:00 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
-----------------------------------------------------------------
CLI:
    Version: 1.2.13.20230704
    Build ID: 00000000

Service:
    Version: 1.2.13.20230704
    Build ID: 00000000
    Level Zero Version: 1.14.0
-----------------------------------------------------------------
  Driver UUID                                     32332e31-372e-3236-3234-312e33330000
  Driver Version                                  23.17.26241.33
-----------------------------------------------------------------
igpu not detected
-----------------------------------------------------------------
xpu-smi is properly installed. 
-----------------------------------------------------------------
+-----------+--------------------------------------------------------------------------------------+
| Device ID | Device Information                                                                   |
+-----------+--------------------------------------------------------------------------------------+
| 0         | Device Name: Intel Corporation Device 56a0 (rev 08)                                  |
|           | Vendor Name: Intel(R) Corporation                                                    |
|           | UUID: 00000000-0000-0003-0000-000856a08086                                           |
|           | PCI BDF Address: 0000:03:00.0                                                        |
|           | DRM Device: /dev/dri/card0                                                           |
|           | Function Type: physical                                                              |
+-----------+--------------------------------------------------------------------------------------+
-----------------------------------------------------------------
```

File: python/llm/scripts/env-check.sh [hunk 1]
```diff
@@ -125,6 +132,42 @@ check_ipex()
...
+  uname -a
+}
+
+check_xpu_driver()
+{
+  echo "-----------------------------------------------------------------"
+  xpu-smi -v
+}
+
+check_OpenCL_driver()
```
Comment (hkvision) [inline python/llm/scripts/env-check.sh:153]: Comment this out since it has sudo as well?

File: python/llm/scripts/env-check.sh [hunk 2]
```diff
@@ -125,6 +132,42 @@ check_ipex()
...
+  sudo clinfo | fgrep Driver
+}
+
+check_igpu()
+{
+  echo "-----------------------------------------------------------------"
+  sycl_ls_output=$(sycl-ls)
+  if echo "$sycl_ls_output" | grep -q "Intel(R) UHD Graphics"; then
+      echo "igpu detected"
+      echo "$sycl_ls_output" | grep "Intel(R) UHD Graphics"
```
Comment (hkvision) [inline python/llm/scripts/env-check.sh:165]: is this correct? igpu should be there

**Comment by hkvision** [comment]
Also need to update the README?

**Comment by hkvision** [comment]
For the CPU information, seems fewer than the original version?

File: python/llm/scripts/env-check.sh (hunk 1 above)
Comment (WeiguangHan) [reply python/llm/scripts/env-check.sh:153]: I tried it again. It didn't need `sudo`.

File: python/llm/scripts/README.md [hunk 3]
```diff
@@ -34,18 +34,20 @@ sudo apt install xpu-smi
...
 -----------------------------------------------------------------
-transformers=4.37.0
+transformers=4.31.0
 -----------------------------------------------------------------
 torch=2.1.0a0+cxx11.abi
 -----------------------------------------------------------------
-IPEX Version: 2.5.0b20240219
+ipex-llm Version: 2.1.0b20240327
 -----------------------------------------------------------------
+/home/arda/anaconda3/envs/new-ipex-llm/lib/python3.9/site-packages/torchvision/io/image.py:13: UserWarning: Failed to load image Python extension: ''If you don't plan on using image functionality from `torchvision.io`, you can ignore this warning. Otherwise, there might be something wrong with your environment. Did you have `libjpeg` or `libpng` installed before building `torchvision` from source?
```
Comment (hkvision) [inline python/llm/scripts/README.md:49]: remove this?

File: python/llm/scripts/README.md [hunk 4]
```diff
@@ -57,52 +59,58 @@ CPU(s):                          32
...
 xpu-smi is properly installed. 
 -----------------------------------------------------------------
 +-----------+--------------------------------------------------------------------------------------+
 | Device ID | Device Information                                                                   |
 +-----------+--------------------------------------------------------------------------------------+
-| 0         | Device Name: Intel(R) Arc(TM) A770 Graphics                                          |
+| 0         | Device Name: Intel Corporation Device 56a0 (rev 08)                                  |
 |           | Vendor Name: Intel(R) Corporation                                                    |
-|           | SOC UUID: 00000000-0000-0003-0000-000856a08086                                       |
+|           | UUID: 00000000-0000-0003-0000-000856a08086                                           |
```
Comment (hkvision) [inline python/llm/scripts/README.md:113]: revert this

Comment (WeiguangHan) [reply python/llm/scripts/README.md]: sure

**Review by hkvision — APPROVED** [review]

**Pull request merged**